sql,roachpb: add plan hash correct value to persisted stats

maryliag opened the pull request:
Previously, we didn't have the plan hash/gist values, so
a dummy value was being used instead. Now that we have the value,
this commit uses those values to be corrected stored.
The Plan Hash is saved on its own column and is part of a
statement key. A plan gist is a string saved in the metadata
and can later on converted back into a logical plan.

Partially addresses #72129

Release note (sql change): Saving plan hash/gist to the Statements
persisted stats.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/75762)
<!-- Reviewable:end -->

Azhng reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75762)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Azhng and @maryliag)

___
*[pkg/roachpb/app_stats.go, line 57 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/75762#-Muqhxv_2n6jngD6a9dh:-Muqhxv_2n6jngD6a9di:beym73x) ([raw file](https://github.com/cockroachdb/cockroach/blob/3da0623017a83304f747ba6e50996ad5092e0bec/pkg/roachpb/app_stats.go#L57)):*
> ```Go
> 		fnv.Add('E')
> 	}
> 	fnv.Add(planHash)
> ```

I think the statement fingerprint ID should be constructed without the `plan_hash`. Otherwise, it would be quite difficult to identify different plans used by the statement fingerprint, e.g.:

``` sql 
SELECT
  statistics ->> 'plan_gist'
FROM
  system.statement_statistics
WHERE
  fingerprint_id = $1
```
___
*[pkg/sql/sqlstats/persistedsqlstats/sqlstatsutil/json_encoding_test.go, line 53 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/75762#-Muqjc1vBGNWhAGyuRW4:-Muqjc1vBGNWhAGyuRW5:b-vk3bl1) ([raw file](https://github.com/cockroachdb/cockroach/blob/3da0623017a83304f747ba6e50996ad5092e0bec/pkg/sql/sqlstats/persistedsqlstats/sqlstatsutil/json_encoding_test.go#L53)):*
> ```Go
>   "vec":         {{.Bool}},
>   "fullScan":    {{.Bool}},
>   "planGists":   {{.StringArray}}
> ```

Hmm, so in the proto definition, you added the `plan_gist` in the `StatementStatistics` message, which is corresponding to the `statistics` column (few lines below). `metadata` column is mapped to `StatementStatisticsKey` protobuf message. So introducing the `planGist` into the `metadata` might feel a bit weird since now the encoder/decoder responsible for the metadata that's not present in the `StatementStatisticsKey` protobuf message.

I think we have two options here, we can either:
1. put `planGist` in the `statistics` column and it can stay in `StatementStatistics` protobuf message
2. put `planGist` in the `metadata` column and it needs to be moved from `StatementStatistics` protobuf message into `StatementStatisticsKey` proto message.


<!-- Sent from Reviewable.io -->

maryliag reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75762)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Azhng)

___
*[pkg/roachpb/app_stats.go, line 57 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/75762#-Muqhxv_2n6jngD6a9dh:-MurSJGU43r0JcdOPzeL:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/3da0623017a83304f747ba6e50996ad5092e0bec/pkg/roachpb/app_stats.go#L57)):*
<details><summary><i>Previously, Azhng (Archer Zhang) wrote…</i></summary><blockquote>

I think the statement fingerprint ID should be constructed without the `plan_hash`. Otherwise, it would be quite difficult to identify different plans used by the statement fingerprint, e.g.:

``` sql 
SELECT
  statistics ->> 'plan_gist'
FROM
  system.statement_statistics
WHERE
  fingerprint_id = $1
```
</blockquote></details>

Done.
___
*[pkg/sql/sqlstats/persistedsqlstats/sqlstatsutil/json_encoding_test.go, line 53 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/75762#-Muqjc1vBGNWhAGyuRW4:-MurSeVEDf-PfPKrcMer:b-cu5rnc) ([raw file](https://github.com/cockroachdb/cockroach/blob/3da0623017a83304f747ba6e50996ad5092e0bec/pkg/sql/sqlstats/persistedsqlstats/sqlstatsutil/json_encoding_test.go#L53)):*
<details><summary><i>Previously, Azhng (Archer Zhang) wrote…</i></summary><blockquote>

Hmm, so in the proto definition, you added the `plan_gist` in the `StatementStatistics` message, which is corresponding to the `statistics` column (few lines below). `metadata` column is mapped to `StatementStatisticsKey` protobuf message. So introducing the `planGist` into the `metadata` might feel a bit weird since now the encoder/decoder responsible for the metadata that's not present in the `StatementStatisticsKey` protobuf message.

I think we have two options here, we can either:
1. put `planGist` in the `statistics` column and it can stay in `StatementStatistics` protobuf message
2. put `planGist` in the `metadata` column and it needs to be moved from `StatementStatistics` protobuf message into `StatementStatisticsKey` proto message.
</blockquote></details>

I decided to go with option 1, so on the next step I can use the builtin to merge stats instead of creating a new one for metadata. And it could be misleading having plan gist as part of something called Key without actually being part of the Key :) 


<!-- Sent from Reviewable.io -->

Azhng reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75762)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Azhng and @maryliag)

___
*[pkg/sql/sqlstats/persistedsqlstats/sqlstatsutil/testutils.go, line 148 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/75762#-MuvWqocD44untqAZjxq:-MuvWqocD44untqAZjxr:bcblewb) ([raw file](https://github.com/cockroachdb/cockroach/blob/f73ebf4a9d33c801649b392c55607e1738da1be8/pkg/sql/sqlstats/persistedsqlstats/sqlstatsutil/testutils.go#L148)):*
> ```Go
> 		val.SetBool(data.Bool)
> 	case reflect.Slice:
> 		for _, randInt := range data.IntArray {
> ```

Hmm I think some additional code here needs to be added to handle the new `.StringArray` type
___
*[pkg/sql/sqlstats/persistedsqlstats/sqlstatsutil/json_encoding_test.go, line 100 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/75762#-MuvWPeY7LWZ6smLAkWp:-MuvWPeY7LWZ6smLAkWq:b-r2af9s) ([raw file](https://github.com/cockroachdb/cockroach/blob/f73ebf4a9d33c801649b392c55607e1738da1be8/pkg/sql/sqlstats/persistedsqlstats/sqlstatsutil/json_encoding_test.go#L100)):*
> ```Go
>          },
>          "nodes": [{{joinInts .IntArray}}],
>          "planGists":   {{.StringArray}}
> ```

Hmm you might need to define a template function here (similar to `joinInts` defined a line above, and you also need to put that callback into the template engine) so the template engine knows to covert golang's string array into a textual json string.


<!-- Sent from Reviewable.io -->

maryliag reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75762)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Azhng)

___
*[pkg/sql/sqlstats/persistedsqlstats/sqlstatsutil/testutils.go, line 148 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/75762#-MuvWqocD44untqAZjxq:-Mv0FqczCpBWwDLY7e_H:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/f73ebf4a9d33c801649b392c55607e1738da1be8/pkg/sql/sqlstats/persistedsqlstats/sqlstatsutil/testutils.go#L148)):*
<details><summary><i>Previously, Azhng (Archer Zhang) wrote…</i></summary><blockquote>

Hmm I think some additional code here needs to be added to handle the new `.StringArray` type
</blockquote></details>

Done
___
*[pkg/sql/sqlstats/persistedsqlstats/testdata/logical_plan_sampling_for_explicit_txn, line 31 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75762#-Mv0FszU5K9fKh5SDca2:-Mv0FszV0lDkBRPLsVk0:b-6gxl3o) ([raw file](https://github.com/cockroachdb/cockroach/blob/946dabc34b9a956214b9868429261b270b8d2a76/pkg/sql/sqlstats/persistedsqlstats/testdata/logical_plan_sampling_for_explicit_txn#L31)):*
> ```
> should-sample-logical-plan db=defaultdb implicitTxn=true fingerprint=SELECT%_
> ----
> true
> ```

@azhng looks like the changes are triggering new sampling of plan. I changed here on the file the places that were failing if kept as false. The addition of plan hash as part of the key could be what is causing the plan getting sampled more often?
___
*[pkg/sql/sqlstats/persistedsqlstats/sqlstatsutil/json_encoding_test.go, line 100 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/75762#-MuvWPeY7LWZ6smLAkWp:-Mv0GFwaCiwvcDoPRrs3:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/f73ebf4a9d33c801649b392c55607e1738da1be8/pkg/sql/sqlstats/persistedsqlstats/sqlstatsutil/json_encoding_test.go#L100)):*
<details><summary><i>Previously, Azhng (Archer Zhang) wrote…</i></summary><blockquote>

Hmm you might need to define a template function here (similar to `joinInts` defined a line above, and you also need to put that callback into the template engine) so the template engine knows to covert golang's string array into a textual json string.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

maryliag committed:
sql,roachpb: add plan hash correct value to persisted stats

Previously, we didn't have the plan hash/gist values, so
a dummy value was being used instead. Now that we have the value,
this commit uses those values to be corrected stored.
The Plan Hash is saved on its own column and is part of a
statement key. A plan gist is a string saved in the metadata
and can later on converted back into a logical plan.
The combined statements groups the value no using the
plan hash as a key, and creates a list of gist executed by
the same fingerprint id.

Partially addresses #72129

Release note (sql change): Saving plan hash/gist to the Statements
persisted stats inside Statistics column.

Azhng reviewed (commented):
> Reviewed 7 of 17 files at r1, 2 of 6 files at r3, 3 of 5 files at r4, 5 of 11 files at r5.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75762)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Azhng and @maryliag)

___
*[pkg/sql/sqlstats/ssmemstorage/ss_mem_storage.go, line 52 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/75762#-Mv5L69eBO3QSZ3zIBtb:-Mv5L69eBO3QSZ3zIBtc:b-2b6dv5) ([raw file](https://github.com/cockroachdb/cockroach/blob/2b0e45cfbfcf5227d36c29db4b05a23d9bf587e9/pkg/sql/sqlstats/ssmemstorage/ss_mem_storage.go#L52)):*
> ```Go
> 	implicitTxn    bool
> 	database       string
> 	planHash       uint64
> ```

Ah this is my bad, I should have left a comment here. The `planKey` here is used to determine if we should tell the optimizer to build a full `EXPLAIN` plan. `planKey` is embedded into the `stmtKey` to form the full key that uniquely identifies a statement stats. In a stats `Container`, there's a plan sample cache and it's updated each time a [plan](https://github.com/cockroachdb/cockroach/blob/259cdd953781fe244ce4439c18322bba56db7944/pkg/sql/sqlstats/ssmemstorage/ss_mem_writer.go#L104) is sampled. 

Let's move the `planHash` into `stmtKey` struct, this is so that plan sampling would still work correctly. 

Hmm perhaps the name is actually quite confusing, since 


| composition   |         what it uniquely identifies | 
|---------------|---------|
|`[anonymizedStmt, failed, implicitTxn, database]` | statement stats of a specific statement fingerprint ID |
|` [anonymizedStmt, failed, implicitTxn, database, transaction fingerprint id]`  | statement stats of a statement fingerprint ID that were part of a specific transaction fingerprint |
| `[anonymizedStmt, failed, implicitTxn, database, planHash]`  | statement stats of a plan of a  specific statement fingerprint ID |
| `[anonymizedStmt, failed, implicitTxn, database, txn fingerprint ID, planHash]` | statement stats of a plan of a specific statement fingerprint ID that were executed as part of a specific transaction fingerprint ID |


Since, `planKey` actually doesn't actually identifies a plan, this name is quite misleading. Since it's used as a heuristics to determine whether a plan sampling should happen, it's probably more fitting to call it something like `sampledPlanKey` or `statementSignature` 
___
*[pkg/sql/sqlstats/persistedsqlstats/testdata/logical_plan_sampling_for_explicit_txn, line 31 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75762#-Mv0FszU5K9fKh5SDca2:-Mv5KVvyCRa-8T29v1nn:b79bvu) ([raw file](https://github.com/cockroachdb/cockroach/blob/946dabc34b9a956214b9868429261b270b8d2a76/pkg/sql/sqlstats/persistedsqlstats/testdata/logical_plan_sampling_for_explicit_txn#L31)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

@azhng looks like the changes are triggering new sampling of plan. I changed here on the file the places that were failing if kept as false. The addition of plan hash as part of the key could be what is causing the plan getting sampled more often?
</blockquote></details>

Hmm this is bad :thinking: . Plan sampling means we are building a full `EXPLAIN` plan, which is really expensive. I think this should be fixable (see comment above) and we should preserve the plan sampling behavior.


<!-- Sent from Reviewable.io -->

maryliag reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75762)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Azhng)

___
*[pkg/sql/sqlstats/ssmemstorage/ss_mem_storage.go, line 52 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/75762#-Mv5L69eBO3QSZ3zIBtb:-MvJnAR04sBUSO3qYTTQ:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/2b0e45cfbfcf5227d36c29db4b05a23d9bf587e9/pkg/sql/sqlstats/ssmemstorage/ss_mem_storage.go#L52)):*
<details><summary><i>Previously, Azhng (Archer Zhang) wrote…</i></summary><blockquote>

Ah this is my bad, I should have left a comment here. The `planKey` here is used to determine if we should tell the optimizer to build a full `EXPLAIN` plan. `planKey` is embedded into the `stmtKey` to form the full key that uniquely identifies a statement stats. In a stats `Container`, there's a plan sample cache and it's updated each time a [plan](https://github.com/cockroachdb/cockroach/blob/259cdd953781fe244ce4439c18322bba56db7944/pkg/sql/sqlstats/ssmemstorage/ss_mem_writer.go#L104) is sampled. 

Let's move the `planHash` into `stmtKey` struct, this is so that plan sampling would still work correctly. 

Hmm perhaps the name is actually quite confusing, since 


| composition   |         what it uniquely identifies | 
|---------------|---------|
|`[anonymizedStmt, failed, implicitTxn, database]` | statement stats of a specific statement fingerprint ID |
|` [anonymizedStmt, failed, implicitTxn, database, transaction fingerprint id]`  | statement stats of a statement fingerprint ID that were part of a specific transaction fingerprint |
| `[anonymizedStmt, failed, implicitTxn, database, planHash]`  | statement stats of a plan of a  specific statement fingerprint ID |
| `[anonymizedStmt, failed, implicitTxn, database, txn fingerprint ID, planHash]` | statement stats of a plan of a specific statement fingerprint ID that were executed as part of a specific transaction fingerprint ID |


Since, `planKey` actually doesn't actually identifies a plan, this name is quite misleading. Since it's used as a heuristics to determine whether a plan sampling should happen, it's probably more fitting to call it something like `sampledPlanKey` or `statementSignature` 
</blockquote></details>

Done
___
*[pkg/sql/sqlstats/persistedsqlstats/testdata/logical_plan_sampling_for_explicit_txn, line 31 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/75762#-Mv0FszU5K9fKh5SDca2:-MvJn6GC7NIkC0n-ecuK:b-933r32) ([raw file](https://github.com/cockroachdb/cockroach/blob/946dabc34b9a956214b9868429261b270b8d2a76/pkg/sql/sqlstats/persistedsqlstats/testdata/logical_plan_sampling_for_explicit_txn#L31)):*
<details><summary><i>Previously, Azhng (Archer Zhang) wrote…</i></summary><blockquote>

Hmm this is bad :thinking: . Plan sampling means we are building a full `EXPLAIN` plan, which is really expensive. I think this should be fixable (see comment above) and we should preserve the plan sampling behavior.
</blockquote></details>

The change in key fixed this issue! 


<!-- Sent from Reviewable.io -->

xinhaoz commented:
___
*[pkg/server/combined_statement_stats.go, line 143 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/75762#-MvJr46UEfU6pUZdtcJL:-MvJr46UEfU6pUZdtcJM:b-m2vrfz) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f1395a52fbef069f707007c5ccd19c309bff3aa/pkg/server/combined_statement_stats.go#L143)):*
> ```Go
> 					FROM crdb_internal.statement_statistics
> 					%s
> 					GROUP BY
> ```

I'm a little confused by the change of query here and might be missing something. We're already grouping by all of these columns in the view. If we're no longer trying to group by the `plan_hash`, I think we should remove that from the view's query. Is that what this new query is supposed to do? Also if we're moving the limit and AOST calause outside its previous function, we have to add it tot he transactions query as well.


<!-- Sent from Reviewable.io -->

maryliag reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75762)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Azhng and @xinhaoz)

___
*[pkg/server/combined_statement_stats.go, line 143 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/75762#-MvJr46UEfU6pUZdtcJL:-MvJu2HA3RHscJIJEsU3:bg4qbfr) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f1395a52fbef069f707007c5ccd19c309bff3aa/pkg/server/combined_statement_stats.go#L143)):*
<details><summary><i>Previously, xinhaoz (Xin Hao Zhang) wrote…</i></summary><blockquote>

I'm a little confused by the change of query here and might be missing something. We're already grouping by all of these columns in the view. If we're no longer trying to group by the `plan_hash`, I think we should remove that from the view's query. Is that what this new query is supposed to do? Also if we're moving the limit and AOST calause outside its previous function, we have to add it tot he transactions query as well.
</blockquote></details>

The view has separate by plan hash, on this endpoint ( which is used on the tables), we're grouping ignoring the plan hash, but on the details page we want to keep separate just like the original view (which will be a new endpoint). So if I change the original view to ignore the plan hash, I won't be able to split them for the details.

Good catch about the Txn, I made the adjustments


<!-- Sent from Reviewable.io -->

Azhng reviewed (approved):
Nice! Just small style nits left. <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 7 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75762#-:-MvP5YVVCHeBySmkKP4O:b-5kqhn4)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @Azhng, @maryliag, and @xinhaoz)

___
*[pkg/server/combined_statement_stats.go, line 94 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/75762#-MvP4e_U0YAqocqGCPAS:-MvP4e_U0YAqocqGCPAT:b-nzww32) ([raw file](https://github.com/cockroachdb/cockroach/blob/ffd359482319ae5aa4bde89bd659d9e64a01c60c/pkg/server/combined_statement_stats.go#L94)):*
> ```Go
> func getFilterAndParams(
> 	start, end *time.Time, limit int64, testingKnobs *sqlstats.TestingKnobs,
> ) (string, string, []interface{}) {
> ```

small (optional) nit: personally I prefer to name the return variables when there are more than two of them in the function. `getFilterAndParams` implies that there are two return vars and one is `filter` and the other one is `param`. Having a name here for each returned vars might clear things up. It's up to you.
___
*[pkg/server/combined_statement_stats.go, line 129 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/75762#-MvP4WQW1uROttMf63xb:-MvP4WQW1uROttMf63xc:b1fwkhb) ([raw file](https://github.com/cockroachdb/cockroach/blob/ffd359482319ae5aa4bde89bd659d9e64a01c60c/pkg/server/combined_statement_stats.go#L129)):*
> ```Go
> 		`SELECT
>         fingerprint_id,
> 				transaction_fingerprint_id,
> ```

small style nit: seems like the vertical indentation is misaligned. (might cuz mixing tabs or space?) 
___
*[pkg/server/combined_statement_stats.go, line 143 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/75762#-MvP5AqADI0D8ANcO8aE:-MvP5AqADI0D8ANcO8aF:b6smr1f) ([raw file](https://github.com/cockroachdb/cockroach/blob/ffd359482319ae5aa4bde89bd659d9e64a01c60c/pkg/server/combined_statement_stats.go#L143)):*
> ```Go
>         aggregated_ts,
> 				aggregation_interval
>       %s`, whereClause, orderAndLimit)
> ```

same minor nits on the indents 


<!-- Sent from Reviewable.io -->

maryliag reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75762)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @Azhng and @xinhaoz)

___
*[pkg/server/combined_statement_stats.go, line 94 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/75762#-MvP4e_U0YAqocqGCPAS:-MvPFNIZ6v96xF2g64uI:by3cv7s) ([raw file](https://github.com/cockroachdb/cockroach/blob/ffd359482319ae5aa4bde89bd659d9e64a01c60c/pkg/server/combined_statement_stats.go#L94)):*
<details><summary><i>Previously, Azhng (Archer Zhang) wrote…</i></summary><blockquote>

small (optional) nit: personally I prefer to name the return variables when there are more than two of them in the function. `getFilterAndParams` implies that there are two return vars and one is `filter` and the other one is `param`. Having a name here for each returned vars might clear things up. It's up to you.
</blockquote></details>

Made some changes, also added comment :) 
___
*[pkg/server/combined_statement_stats.go, line 129 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/75762#-MvP4WQW1uROttMf63xb:-MvPF4ZuAn1pTozQ5x9B:b-7ql1td) ([raw file](https://github.com/cockroachdb/cockroach/blob/ffd359482319ae5aa4bde89bd659d9e64a01c60c/pkg/server/combined_statement_stats.go#L129)):*
<details><summary><i>Previously, Azhng (Archer Zhang) wrote…</i></summary><blockquote>

small style nit: seems like the vertical indentation is misaligned. (might cuz mixing tabs or space?) 
</blockquote></details>

fixed
___
*[pkg/server/combined_statement_stats.go, line 143 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/75762#-MvP5AqADI0D8ANcO8aE:-MvPF9PF3QhOGnc0oFds:b-7ql1td) ([raw file](https://github.com/cockroachdb/cockroach/blob/ffd359482319ae5aa4bde89bd659d9e64a01c60c/pkg/server/combined_statement_stats.go#L143)):*
<details><summary><i>Previously, Azhng (Archer Zhang) wrote…</i></summary><blockquote>

same minor nits on the indents 
</blockquote></details>

fixed


<!-- Sent from Reviewable.io -->

xinhaoz commented:
___
*[pkg/server/combined_statement_stats.go, line 143 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/75762#-MvJr46UEfU6pUZdtcJL:-MvPHHQs33umJ5RA7yKv:b-d0kdud) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f1395a52fbef069f707007c5ccd19c309bff3aa/pkg/server/combined_statement_stats.go#L143)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

The view has separate by plan hash, on this endpoint ( which is used on the tables), we're grouping ignoring the plan hash, but on the details page we want to keep separate just like the original view (which will be a new endpoint). So if I change the original view to ignore the plan hash, I won't be able to split them for the details.

Good catch about the Txn, I made the adjustments
</blockquote></details>

Ah I see. Thanks for the explanation!


<!-- Sent from Reviewable.io -->

maryliag commented:
TFTR
bors r+
